src/kv/committable_tx.h

```diff
@@ -80,9 +90,12 @@ namespace kv
      * @return transaction outcome
      */
     CommitResult commit(
+      const ccf::ClaimsDigest& claims = ccf::no_claims(),
       bool track_read_versions = false,
       std::function<std::tuple<Version, Version>(bool has_new_map)>
-        version_resolver = nullptr)
+        version_resolver = nullptr,
+      std::function<void(const std::vector<uint8_t>& write_set)>
+        write_set_observer = nullptr)
```

Comment:
If we changed the snapshot generation logic to go through a new node-only endpoint to record the snapshot evidence (rather than calling `tx.commit()` directly), then this extra argument wouldn't be necessary as we would be using the generic mechanism to record the write set as a claim? 

I'm not saying it's necessarily a better solution, but it seems that we have already followed this pattern for data recorded by the current primary node directly (e.g. ORCs).